### PR TITLE
Droth 2426 lanes api error

### PR DESCRIPTION
**Pull request by FilPinto**

### CODE REVIEW

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/lane/LaneService.scala [hunk 1]
```diff
@@ -536,59 +536,49 @@ trait LaneOperations {
     * @param username Username of the user is doing the changes
     * @return
     */
-  def update ( newIncomeLane: Seq[NewIncomeLane], linkIds: Set[Long] ,sideCode: Int, username: String ): Seq[Long] = {
-    if (newIncomeLane.isEmpty || linkIds.isEmpty)
-      return Seq()
+  def update(newIncomeLane: Seq[NewIncomeLane], linkIds: Set[Long], sideCode: Int, username: String, newTransaction: Boolean = true): Seq[Long] = {
+    def updateProcess(): Seq[Long] = {
+      if (newIncomeLane.isEmpty || linkIds.isEmpty)
```
Comment (ricardofmsimoes) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/lane/LaneService.scala:541]: Maybe we can do this validation before if (newTransaction)

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/lane/LaneService.scala [hunk 2]
```diff
@@ -624,12 +614,10 @@ trait LaneOperations {
...
-  def create(newIncomeLane: Seq[NewIncomeLane], linkIds: Set[Long] ,sideCode: Int, username: String, vvhTimeStamp: Long = vvhClient.roadLinkData.createVVHTimeStamp()): Seq[Long] = {
-
-    if (newIncomeLane.isEmpty || linkIds.isEmpty)
-      return Seq()
-
-    withDynTransaction {
+  def create(newIncomeLane: Seq[NewIncomeLane], linkIds: Set[Long] ,sideCode: Int, username: String, vvhTimeStamp: Long = vvhClient.roadLinkData.createVVHTimeStamp(), newTransaction: Boolean = true): Seq[Long] = {
+    def createProcess(): Seq[Long] = {
+      if (newIncomeLane.isEmpty || linkIds.isEmpty)
+        return Seq()
```
Comment (ricardofmsimoes) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/lane/LaneService.scala:620]: Maybe we can do this validation before if (newTransaction)

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/lane/LaneService.scala (hunk 1 above)
Comment (FilPinto) [reply digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/lane/LaneService.scala:541]: Done

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/lane/LaneService.scala (hunk 2 above)
Comment (FilPinto) [reply digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/lane/LaneService.scala:620]: Done

**Pull request merged**